assets/js/modules/analytics/datastore/tags.js

```diff
@@ -197,11 +198,15 @@ export const resolvers = {
 	*getExistingTag() {
 		try {
 			const registry = yield actions.getRegistry();
-			const existingTag = yield actions.fetchExistingTag( 'analytics' );
-			yield actions.receiveExistingTag( existingTag );
 
-			// Invalidate this resolver so it will run again.
-			yield registry.getActions().invalidateResolutionForStoreSelector( 'getExistingTag' );
+			// If an existing tag was already loaded; return early and don't
+			// dispatch the fetch action.
+			if ( registry.select( STORE_NAME ).getExistingTag() !== undefined ) {
+				return;
+			}
+
+			const existingTag = yield actions.fetchExistingTag();
```

Comment:
If might be nicer to change the `fetchExistingTag` action/control to return `null` itself if there isn't a tag returned. Then we wouldn't need this check, because I don't think we should be returning `undefined` from `fetchExistingTag()` in general anyway—`null` seems like a more appropriate return value if nothing is available.

But I get that's a separate API and we shouldn't refactor it right now. For now—let's just put the `existingTag !== undefined ? existingTag : null` check in our own `fetchExistingTag()` action instead of here in the resolver.

Reply:
I've opened #1328 to implement a datastore-proper mechanism for tag detection.